backend/routes/queries.py

```
@@ -0,0 +1,342 @@
+            client_ip=row["client_ip"],
+            location_city=row["location_city"],
+            location_region=row["location_region"],
+            location_country=row["location_country"],
+            location_country_code=row["location_country_code"],
+        )
+
+    except HTTPException:
+        raise
+    except sqlite3.Error as e:
+        logger.error(f"Database error in get_query: {e}")
+        raise HTTPException(status_code=500, detail="Database error")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)